common/src/main/java/com/wynntils/core/webapi/profiles/item/ItemType.java

```diff
@@ -5,28 +5,59 @@
 package com.wynntils.core.webapi.profiles.item;
 
 import com.wynntils.wc.objects.ClassType;
+import java.util.Locale;
 
 public enum ItemType {
-    SPEAR(ClassType.Warrior),
-    WAND(ClassType.Mage),
-    DAGGER(ClassType.Assassin),
-    BOW(ClassType.Archer),
-    RELIK(ClassType.Shaman),
-    RING(null),
-    BRACELET(null),
-    NECKLACE(null),
-    HELMET(null),
-    CHESTPLATE(null),
-    LEGGINGS(null),
-    BOOTS(null);
+    SPEAR(ClassType.Warrior, 16 * 1, 16 * 1),
+    WAND(ClassType.Mage, 16 * 0, 16 * 1),
+    DAGGER(ClassType.Assassin, 16 * 2, 16 * 1),
+    BOW(ClassType.Archer, 16 * 3, 16 * 1),
+    RELIK(ClassType.Shaman, 16 * 0, 16 * 2),
+    RING(null, 16 * 1, 16 * 2),
+    BRACELET(null, 16 * 2, 16 * 2),
+    NECKLACE(null, 16 * 3, 16 * 2),
+    HELMET(null, 16 * 0, 16 * 0),
+    CHESTPLATE(null, 16 * 1, 16 * 0),
+    LEGGINGS(null, 16 * 2, 16 * 0),
+    BOOTS(null, 16 * 3, 16 * 0);
 
     private final ClassType classReq;
+    private final int iconTextureX;
+    private final int iconTextureY;
 
-    ItemType(ClassType classReq) {
+    ItemType(ClassType classReq, int iconTextureX, int iconTextureY) {
         this.classReq = classReq;
+        this.iconTextureX = iconTextureX;
+        this.iconTextureY = iconTextureY;
     }
 
     public ClassType getClassReq() {
         return classReq;
     }
+
+    public int getIconTextureX() {
+        return iconTextureX;
+    }
+
+    public int getIconTextureY() {
+        return iconTextureY;
+    }
+
+    public static ItemType fromString(String typeStr) {
+        return switch (typeStr.toLowerCase(Locale.ROOT)) {
+            case "spear", "hammer", "scythe", "warrior", "knight", "wa" -> SPEAR;
```

Comment:
Are we ever going to get any other than the first? Like "Unidentified Scythe" -- is that a thing?

I see you have copied this from legacy. I'm just surprised, this was not how I left it when I created the item type overlays in legacy...

Reply:
I see. Seems I just used a pre-existing function, which did what I needed (and more). Not sure we need to carry that baggage over to Artemis.

Reply:
I copied it from legacy. Although it might be feature envy to add the other cases without a feature that uses them.

Reply:
I'd say remove all string literals except the ones we need.